modin/experimental/xgboost/xgboost.py

```diff
@@ -80,6 +119,170 @@ def __iter__(self):
+        }
+        return dmatrix_params
+
+    @property
+    def feature_names(self):
+        """
+        Get column labels.
+
+        Returns
+        -------
+        Column labels.
+        """
+        return self._feature_names
+
+    @feature_names.setter
+    def feature_names(self, feature_names):
+        """
+        Set column labels.
+
+        Parameters
+        ----------
+        feature_names : list or None
+            Labels for columns. In the case of ``None``, existing feature names will be reset.
+        """
+        if feature_names is not None:
+            feature_names = (
+                list(feature_names)
+                if not isinstance(feature_names, str)
+                else [feature_names]
+            )
+
+            if len(feature_names) != len(set(feature_names)):
+                raise ValueError("Items in `feature_names` must be unique.")
+            if len(feature_names) != self.num_col() and self.num_col() != 0:
+                raise ValueError(
+                    "`feature_names` must have the same width as `self.data`."
+                )
+            if not all(
+                isinstance(f, str) and not any(x in f for x in set(("[", "]", "<")))
+                for f in feature_names
+            ):
+                raise ValueError(
+                    "Items of `feature_names` must be string and must not contain [, ] or <."
+                )
+        else:
+            feature_names = None
+        self._feature_names = feature_names
+
+    @property
+    def feature_types(self):
+        """
+        Get column types.
+
+        Returns
+        -------
+        Column types.
+        """
+        return self._feature_types
+
+    @feature_types.setter
+    def feature_types(self, feature_types):
+        """
+        Set column types.
+
+        Parameters
+        ----------
+        feature_types : list or None
+            Labels for columns. In case None, existing feature names will be reset.
+        """
+        if feature_types is not None:
+            if not isinstance(feature_types, (list, str)):
+                raise TypeError("feature_types must be string or list of strings")
+            if isinstance(feature_types, str):
+                feature_types = [feature_types] * self.num_col()
+                feature_types = (
+                    list(feature_types)
+                    if not isinstance(feature_types, str)
+                    else [feature_types]
+                )
+        else:
+            feature_types = None
+        self._feature_types = feature_types
+
+    def num_row(self):
+        """
+        Get number of rows.
+
+        Returns
+        -------
+        int
+        """
+        return self._n_rows
+
+    def num_col(self):
+        """
+        Get number of columns.
+
+        Returns
+        -------
+        int
+        """
+        return self._n_cols
+
+    def get_float_info(self, name):
```

Comment:
Is this method really implemented appropriately? I see native XGBoost has some implementation details for the method.

Reply:
This is analog with woking in original XGBoost